features/admin.users.v1/components/user-impersonation-action.tsx

```diff
@@ -169,6 +171,25 @@ export const UserImpersonationAction: FunctionComponent<UserImpersonationActionI
         }
     }, [ idToken, subjectToken ]);
 
+    useEffect(() => {
+        if (impersonationInProgress) {
+            setTimeout(() => {
+                if (impersonationInProgress && idToken == undefined && subjectToken == undefined) {
+                    setImpersonationInProgress(false);
+                    dispatch(addAlert({
+                        description: t(
+                            "users:notifications.impersonateUser.error.description"
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: t(
+                            "users:notifications.impersonateUser.error.message"
+                        )
+                    }));
+                }
+            }, 5000);
```

Comment:
why do we need a setTimeout here?

Reply:
On an unexpected error occurred during SSO, no way to detect that as the error is in the other iframe different host. Because of that, for better UX, the impersonation action will fail after 5 sec.

Reply:
Ah okay, now only I remembered. Better if we could add a comment.

Reply:
Updated the PR description.